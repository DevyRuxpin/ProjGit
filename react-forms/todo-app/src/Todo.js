import React from 'react';

function Todo({ id, task, removeTodo }) {
  const handleRemove = () => {
    removeTodo(id);
  };

  return (
    <div className="Todo">
      <span>{task}</span>
      <button onClick={handleRemove}>X</button>
    </div>
  );
}

export default Todo; 