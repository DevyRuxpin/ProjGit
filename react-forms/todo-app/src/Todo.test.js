import { render, screen, fireEvent } from '@testing-library/react';
import Todo from './Todo';

const mockRemoveTodo = jest.fn();

test('renders without crashing', () => {
  render(<Todo id={1} task="Buy groceries" removeTodo={mockRemoveTodo} />);
});

test('matches snapshot', () => {
  const { asFragment } = render(<Todo id={1} task="Buy groceries" removeTodo={mockRemoveTodo} />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders task text', () => {
  render(<Todo id={1} task="Buy groceries" removeTodo={mockRemoveTodo} />);
  expect(screen.getByText('Buy groceries')).toBeInTheDocument();
});

test('renders remove button', () => {
  render(<Todo id={1} task="Buy groceries" removeTodo={mockRemoveTodo} />);
  expect(screen.getByText('X')).toBeInTheDocument();
});

test('calls removeTodo when X button is clicked', () => {
  render(<Todo id={1} task="Buy groceries" removeTodo={mockRemoveTodo} />);
  
  const removeButton = screen.getByText('X');
  fireEvent.click(removeButton);
  
  expect(mockRemoveTodo).toHaveBeenCalledWith(1);
}); 