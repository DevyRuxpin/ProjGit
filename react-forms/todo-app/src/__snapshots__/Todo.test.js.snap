// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <span>
      Buy groceries
    </span>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
